programs/bpf_loader/src/lib.rs

```diff
@@ -55,6 +56,10 @@ pub const DEFAULT_LOADER_COMPUTE_UNITS: u64 = 570;
 pub const DEPRECATED_LOADER_COMPUTE_UNITS: u64 = 1_140;
 pub const UPGRADEABLE_LOADER_COMPUTE_UNITS: u64 = 2_370;
 
+thread_local! {
+    pub static MEMORY_POOL: RefCell<VmMemoryPool> = RefCell::new(VmMemoryPool::new());
```

Comment:
Does this mean we allocate a `VmMemoryPool` for every thread in the system?

Reply:
Yes, so we don't need any kind of locking. This allocates 2.5MB per thread. Before we were already caching these allocations per thread in jemalloc, now we just do it explicitly in our code.

Reply:
My mnb validator has 579 threads, that's over 1.4GB. Very few of those will actually be used.

Reply:
yes, you're right, nvm